src/instructlab/sdg/blocks/iterblock.py

```diff
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+import logging
+
+# Third Party
+from datasets import Dataset
+
+# Local
+from ..pipeline import _lookup_block_type
+from ..registry import BlockRegistry
+from .block import Block
+
+logger = logging.getLogger(__name__)
+
+
+# This is part of the public API.
+@BlockRegistry.register("IterBlock")
+class IterBlock(Block):
```

Comment:
Could we add a docstring to explain what this new block does? It would be helpful to briefly describe its purpose, key attributes like `num_iters`, and how it fits into the overall system. This will make it easier for others to understand its functionality and use. We also don't have to add it in this PR, just a recommendation.

Reply:
Done!